pkgs/development/interpreters/python/cpython/3.6/default.nix

```diff
@@ -64,6 +64,14 @@ in stdenv.mkDerivation {
     substituteInPlace configure --replace '-Wl,-stack_size,1000000' ' '
   '';
 
+  patches = optionals (!(stdenv.cc.isGNU or false)) [
```

Comment:
I generally prefer to minimize amount of "branchy" logic between platforms, if we can pull it off. I'd make this unconditional, adding a comment saying what we say here, i.e., that gcc is okay with it and clang isn't, and we're doing this so we don't bake in assumptions about weird behavior in specific compilers.